wavelength-plugins/friends.js

```diff
@@ -0,0 +1,68 @@
+/****************************
+ * Friend List
+ * Original idea from WGC
+ * Coded by Volco
+ * *************************/
+'use strict';
+
+function onlineFriends(user) {
+	let friends = Db.friends.get(user);
+	let list = [];
+	friends.forEach(friend => {
+		if (Users(friend).connected) list.push(friend);
+	});
+	if (list.length) user.send(`|pm| WL Friend Manager|You have ${list.length} friends online: ${list.join(', ')}`);
+}
+WL.onlineFriends = onlineFriends;
+
+exports.commands = {
+	friend: 'friends',
+	friends: {
+		add: function (target, room, user) {
+			if (!target) return this.parse(`/help friends add`);
+			let targetUser = toId(target);
+			if (user.userid === targetUser) return this.errorReply(`You cannot add yourself!`);
+			if (targetUser.length < 1 || targetUser.length > 19) return this.errorReply(`Names should be more than 1 character and less than 19 characters long.`);
+			if (Db.friends.get(user.userid, []).indexOf(targetUser) !== -1) return this.errorReply(`You already added them!`);
```

Comment:
Prefer `Db.friends.get(user.userid, []).includes(targetUser)`.

Reply:
The reason i use indexOf is because say someone is named a and then you have a friend named abc abc includes a (just an example i've seen similar named before that happen to be a small part of a different users name this is just to prevent glitches)

Reply:
That shouldn't ever happen with arrays

Reply:
ok